corehq/ex-submodules/casexml/apps/phone/tasks.py

```diff
@@ -37,7 +38,16 @@ def force_update_cleanliness_flags():
 def get_async_restore_payload(restore_config):
     """Process an async restore
     """
-    response = restore_config.generate_payload(async_task=current_task)
+    status = 'err'
+    try:
+        with restore_config.timing_context:
+            response = restore_config.generate_payload(async_task=current_task)
+        status = 200
+    except RestoreException:
+        status = 412
+        raise
+    finally:
+        record_restore_timing(restore_config, status)
```

Comment:
good question. I could probably do that, but it's going to take a bit of engineering. Can we merge this PR and I'll implement that separately?